### DISCUSSION TITLE
Wire target using multiple function with parameter, seems not working?

### DISCUSSION THREAD

**Discussion by whousehomestay** [livewire/livewire#8134] [7. Bugs (archived)] [answered]
### Livewire version

v3.4.9

### Laravel version

v10.48.3

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

Use something like:

wire:target="nextPage('paginator-name'),previousPage('paginator-name')"

will produce error on the console:

```
Uncaught ReferenceError: previousPage is not defined
    at eval (eval at parseOutMethodAndParams (livewire.js?id=5d8beb2e:4666:20), <anonymous>:8:29)
    at Directive.parseOutMethodAndParams (livewire.js?id=5d8beb2e:4672:18)
    at get method (livewire.js?id=5d8beb2e:4653:31)
    at getTargets (livewire.js?id=5d8beb2e:9293:43)
    at livewire.js?id=5d8beb2e:9175:33
    at Array.<anonymous> (livewire.js?id=5d8beb2e:4610:9)
    at trigger2 (livewire.js?id=5d8beb2e:3857:34)
    at livewire.js?id=5d8beb2e:8437:11
    at Array.forEach (<anonymous>)
    at livewire.js?id=5d8beb2e:8436:21
```

### Repository or Wirebox URL

-

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by whousehomestay** [comment]
I don't know the problem is on wire:target or on pagination itself.

**Comment by danie-ramdhani** [comment]
mmmm... yeah it's bug. 

https://wirebox.app/b/xeov6

in the wirebox example, the `bar` will evaluated as a variable which is not exist.

![image](https://github.com/livewire/livewire/assets/58886605/eb995004-8ec6-4e5a-acad-04360f4f6a67)

maybe you can try workaround to not using targeted actions (`wire:target="nextPage, previousPage"`) or using `@foreach`

```blade
@foreach (["nextPage('paginator-name')", "previousPage('paginator-name')"] as $item)
    <div wire:loading wire:target="{{ $item }}">Loading</div>
@endforeach
```

**Reply by whousehomestay** [reply]
Yeah, maybe "foreach" (and it means duplicating same element many times) is the temporary solution. My problem is because I use multiple paginations on a page, and I need to use targeted action to differentiate loading indicator for each pagination.

**Reply by danie-ramdhani** [reply]
made a failing test: https://github.com/livewire/livewire/pull/8159

**Reply by danie-ramdhani** [reply, marked as answer]
it's not supported yet to add multiple action parameters https://github.com/livewire/livewire/pull/8159#issuecomment-2022986030

**Comment by patriktoth67** [comment]
The issue is in 'wire-loading.js' in the getTargets function I believe:

```
        if (raw.includes('(') && raw.includes(')')) {
            targets.push({ target: directive.method, params: quickHash(JSON.stringify(directive.params)) })
        } else if (raw.includes(',')) {
            raw.split(',').map(i => i.trim()).forEach(target => {
                targets.push({ target })
            })
        } else {
            targets.push({ target: raw })
        }
 ```
 
 This code only covers if there is one method target, which can have multiple parameters like: `wire:target="bar('baz', 'qux')"`
 
As soon as you add another method (like `wire:target="foo('bar'), bar('foo')`) it won't work, since it cannot decide what's the method and the params.

Anyone doing a PR for this, I'd start there.
